spring-cloud-data-core/src/main/java/org/springframework/cloud/data/core/TaskDefinition.java

```diff
@@ -0,0 +1,222 @@
+/*
+ * Copyright 2015 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.cloud.data.core;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.springframework.cloud.data.core.parser.TaskDefinitionParser;
+import org.springframework.core.style.ToStringCreator;
+import org.springframework.util.Assert;
+
+/**
+ * @author Michael Minella
+ */
+public class TaskDefinition {
+
+	private static final TaskDefinitionParser parser = new TaskDefinitionParser();
+
+	/**
+	 * Name of module.
+	 */
+	private final String taskName;
+
+	/**
+	 * Symbolic taskName of a module. This may be generated to a default
+	 * value or specified in the DSL string.
+	 */
+	private final String task;
+
+	/**
+	 * Parameters for module. This is specific to the type of module - for
+	 * instance an http module would include a port number as a parameter.
+	 */
+	private final Map<String, String> parameters;
+
+
+	/**
+	 * Construct a {@code TaskDefinition}. This constructor is private; use
+	 * {@link TaskDefinition.Builder} or
+	 * {@link TaskDefinition#getInstance(String, String)} to create a new instance.
+	 *
+	 * @param taskName taskName of task
+	 * @param task task to be used
+	 * @param parameters module parameters; may be {@code null}
+	 */
+	private TaskDefinition(String taskName, String task, Map<String, String> parameters) {
```

Comment:
we should be delegating to the parser within the constructor to be consistent with StreamDefinition, unless there is a reason this is different?